setup/install_gatling/99_final/99_final_puppet_run.rb

```diff
@@ -4,4 +4,12 @@
   step "Run puppet until all changes are applied" do
     run_agent_until_no_change(hosts)
   end
+  step "Ensure puppet agent disabled" do
+    # If deployed via install_pe this is part of the install.
+    #   Ref: https://github.com/puppetlabs/beaker-pe/blob/master/lib/beaker-pe/install/pe_utils.rb
+    # This is added to ensure that the puppet agent service is
+    # disabled prior to a gatling run regardless of how the hosts
+    # were deployed.
+    stop_agent_on(hosts)
```

Comment:
So what if we create an actual agent?  We have a rake task for that.  Should this be more selective?  Not even sure there is a good way to do that... so just asking the question really. :)

Reply:
Answer: Confine the step to hosts that do not include any of the PE Infra beaker roles.
```
pe_infra_roles = ["master", "database", "dashboard", "compile_master", "metric"]
confine :to, {}, hosts.select { |host| ! (host & pe_infra_roles).any? }
```

Reply:
![Super beaker](https://vignette.wikia.nocookie.net/muppet/images/2/2f/Superbeaker01.jpg/revision/latest/scale-to-width-down/300?cb=20060516023018)

Reply:
looks good.  go for it.

Reply:
This has been added and verified.  I had the inclusion logic inverted in my comment above.  This has been corrected in the code.